core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -394,12 +397,20 @@ private TaskInfo doUpdateTask(
         SqlTask sqlTask = tasks.getUnchecked(taskId);
         QueryContext queryContext = sqlTask.getQueryContext();
         if (!queryContext.isMemoryLimitsInitialized()) {
-            long sessionQueryMaxMemoryPerNode = getQueryMaxMemoryPerNode(session).toBytes();
+            RetryPolicy retryPolicy = getRetryPolicy(session);
+            if (retryPolicy == TASK) {
+                // Memory limit for fault tolerant queries is enforced by the MemoryPool.
```

Comment:
What do you mean by "Memory limit for fault tolerant queries is enforced by the MemoryPool"?

Reply:
I meant "Memory limit for fault tolerant queries should be enforced only by the MemoryPool". Let me rephrase it.